code/modules/modular_computers/computers/item/computer.dm

```diff
@@ -280,9 +280,8 @@ GLOBAL_LIST_EMPTY(TabletMessengers) // a list of all active messengers, similar
 	if(crew_manifest_update)
 		GLOB.manifest.modify(computer_id_slot.registered_name, computer_id_slot.assignment, computer_id_slot.get_trim_assignment())
 
-	if(user)
-		if(!issilicon(user) && in_range(src, user))
-			user.put_in_hands(computer_id_slot)
+	if(user && !issilicon(user) && in_range(src, user))
+		user.put_in_hands(computer_id_slot)
 		balloon_alert(user, "removed ID")
 		to_chat(user, span_notice("You remove the card from the card slot."))
```

Comment:
We should still send these messages if a silicon ejected the ID, it's just that the following line (`whatever.forcemove`) should always happen even if we don't have a user

Reply:
👍